doc/getting-started/roscube/roscube-gsg.rst

```diff
@@ -0,0 +1,720 @@
+
+   .. code-block:: bash
+
+     sudo update-grub
+
+#. Poweroff the VM.
+
+   .. code-block:: bash
+
+     sudo poweroff
+
+Run real-time VM
+================
+
+Now back to the native machine and we'll set up the environment for
+launching the real-time VM.
+
+#. Convert KVM image file format.
+
+   .. code-block:: bash
+
+     mkdir -p ~/acrn/rtosVM
+     cd ~/acrn/rtosVM
+     sudo qemu-img convert -f qcow2 \
+       -O raw /var/lib/libvirt/images/ROS2SystemRTOS.qcow2 \
+       ./ROS2SystemRTOS.img
+
+#. Create a new launch file
+
+   .. code-block:: bash
+
+     wget https://raw.githubusercontent.com/Adlink-ROS/ROScube_ACRN_guide/master/scripts/launch_ubuntu_rtos.sh
+     chmod +x ./launch_ubuntu_rtos.sh
+
+#. **Reboot to ACRN kernel** and now you can launch the VM.
+
+   .. code-block:: bash
+
+     cd ~/acrn/rtosVM
+     sudo ./launch_ubuntu_rtos.sh
+
+.. note:: Please use ``poweroff`` instead of ``reboot`` in the real-time VM.
+   In ACRN design, the system will also reboot while rebooting real-time VM.
+
+About the Launch File
+*********************
+
+The launch file in this tutorial has the following hardware resource allocation.
+
+.. csv-table::
+   :header: "Resource", "Service VM", "User VM", "Real-time VM"
+   :widths: 15, 15, 15, 15
+
+   "CPU", "0", "1,2,3", "4,5"
+   "Memory", "Remaining", "8 GB", "2 GB"
+   "Ethernet", "Ethernet 1 & 2", "Ethernet 3", "Ethernet 4"
+   "USB", "Remaining", "1-2", "1-1"
+
+You can modify the launch file for your own hardware resource allocation.
+We'll provide some modification methods below.
+For more detail, please refer to `Device Model Parameters
+<https://projectacrn.github.io/latest/user-guides/acrn-dm-parameters.html>`_.
+
+CPU
+===
+
+Modify the number behind ``--cpu-affinity`` in the command acrn-dm.
+The number should be from 0 to max CPU ID.
+For example, if you want to run VM with core 1 and 2, use ``--cpu-affinity 1,2``.
+
+Memory
+======
+
+Modify the ``mem_size`` in launch file. This variable will be passed to acrn-dm.
+The possible values are 1024M, 2048M, 4096M, 8192M.
+
+Ethernet
+========
+
+Run ``lspci -Dnn | grep "Ethernet controller"`` to get the ID of Ethernet port.
+
+.. figure:: images/rqi-acrn-ethernet-lspci.png
+
+You'll see 4 ID for each Ethernet port.
+Assign the ID of the port you want to passthrough in the launch file.
+For example, if we want to passthrough Ethernet 3 to the VM:
+
+.. code-block:: bash
+
+  passthru_vpid=(
+  ["ethernet"]="8086 1539"
+  )
+  passthru_bdf=(
+  ["ethernet"]="0000:04:00.0"
+  )
+
+  # Passthrough ETHERNET
+  echo ${passthru_vpid["ethernet"]} > /sys/bus/pci/drivers/pci-stub/new_id
+  echo ${passthru_bdf["ethernet"]} > /sys/bus/pci/devices/${passthru_bdf["ethernet"]}/driver/unbind
+  echo ${passthru_bdf["ethernet"]} > /sys/bus/pci/drivers/pci-stub/bind
+
+  acrn-dm
+  ⋮
+  -s 4,passthru,04/00/0 \
+  ⋮
+
+USB
+===
+
+To passthrough USB to VM, we need to know the ID for each USB first.
+
+.. figure:: images/rqi-acrn-usb-port.png
```

Comment:
The USB port number do not match what's printed on the enclosure, I'm assuming this is the way it is though, isn't it?

Reply:
Yes, we need to use the number on the picture to passthrough the USB. This number is different from the number on ROScube panel.

Reply:
Thanks for the confirmation.